### PR TITLE
CEM-951 Box Component

### PR DESCRIPTION
**Pull request by Ampatte2**
Component from terminal, made more extensible, used theme.

CEM-951-Box-Component-Selected
CEM-951-Box-Component

### CODE REVIEW

File: src/Containers/Box.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import styled from 'styled-components';
+
+export interface BoxProps {
```
Comment (ralph-dev) [inline src/Containers/Box.tsx:4]: does this not extend div Interface?

**Review by ralph-dev — APPROVED** [review]
LGTM, except for the possible missing extension of interface?

File: src/Containers/Box.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            {children}
+        </BoxDiv>
+    );
+};
+
+const BoxDiv = styled.div<BoxProps>`
+    ${({ theme, ...props }): string => `
+        box-shadow: ${theme.depth[2]};
+        border-radius: ${theme.dimensions.radius};
+        background-color: ${props.selected && theme.colors.primary};
```
Comment (fpunny) [inline src/Containers/Box.tsx:33]: tbh when would it be selected? what does it mean for a generic box to be selected? isnt it better just to have a color prop?

Comment (fpunny) [reply src/Containers/Box.tsx:33]: and if it does make sense, then i guess box is a really bad name for this

File: src/Containers/SelectBox.tsx [hunk 3]
```diff
@@ -1,32 +1,34 @@
 import React from 'react';
 import styled from 'styled-components';
 
-export interface BoxProps {
+export interface SelectBoxProps extends React.HTMLAttributes<HTMLDivElement> {
     text?: string;
     align?: string;
-    selected?: boolean;
+    selected?: string;
```
Comment (ralph-dev) [inline src/Containers/SelectBox.tsx]: isn't this a boolean?

Comment (ralph-dev) [reply src/Containers/SelectBox.tsx]: isSelected if it is a boolean btw

Comment (Ampatte2) [reply src/Containers/SelectBox.tsx]: commit Formatting and prop type fixes that

**Comment by Ampatte2** [comment]
Changed to isSelected

**Review by ralph-dev — APPROVED** [review]
LGTM, bold should be isBold though

**Comment by Ampatte2** [comment]
isBold would be inconsistent with the other current implementations in the library.

**Comment by ralph-dev** [comment]
> isBold would be inconsistent with the other current implementations in the library.

Ok then keep it for now, but according to the guide it's wrong haha. I will write an automated script one day to fix all the misnamed booleans

**Pull request merged**